server/api/utils/clients/async_nuclio.py

```diff
@@ -55,13 +120,18 @@ async def _ensure_async_session(self):
                 logger=logger,
             )
 
+    async def close_session(self):
+        if self._session:
+            await self._session.close()
+            self._session = None
```

Comment:
use some sort of `@contextlib.asynccontextmanager` functionality. better and simplify the boilerplating